image_writers_guide/guidelines.adoc

```diff
@@ -5,143 +5,186 @@
+////
+
+NOTE: Even with persisted volumes, each instance of your image has its own volume, and the filesystem is not shared between instances.  This means the volume cannot be used to share state in a cluster.
+
+== OpenShift-Specific Guidelines
+The following are guidelines that apply when creating Docker images specifically for use on OpenShift.
+
+*Enabling Images for Source-To-Image (STI)*
+
+For images that are intended to run application code provided by a third party, such as a Ruby image designed to run Ruby code provided by a developer, you can enable your image to work with the https://github.com/openshift/source-to-image[Source-to-Image (STI)]  build tool. STI is a framework which makes it easy to write images that take application source code as an input and produce a new image that runs the assembled application as output.
+
+For example, this https://github.com/openshift/wildfly-8-centos[Wildfly image] defines STI scripts which run a `maven` build on a Java source repository and copy the resulting [sysitem]#war# file into the Wildfly deployments directory. The resulting image now automatically starts Wildfly with the application running.
+
+For more details about how to write STI scripts for your image, see the link:sti.html[STI Requirements] topic.
+
+[[using-services]]
+*Using Services for Inter-image Communication*
+
+For cases where your image needs to communicate with a service provided by another image, such as a web front end image that needs to access a database image to store and retrieve data, your image should consume an OpenShift link:../architecture/kubernetes_model.html#service[service]. Services provide a static endpoint for access which does not change as containers are stopped, started, or moved. In addition, services provide load balancing for requests.
 
-Keep in mind that even with persisted volumes, each instance of your image will have its own volume, the filesystem is not shared between instances.  This means the volume cannot be used to share state in a cluster.
+////
+For more information see https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/services.md[this documentation].  (NOTE to docs team:  this link should really go to something in the openshift docs once we have it)
+////
 
+*Providing Common Libraries*
 
+For images that are intended to run application code provided by a third party, ensure that your image contains commonly used libraries for your platform. In particular, provide database drivers for common databases used with your platform. For example, provide JDBC drivers for MySQL and PostgreSQL if you are creating a Java framework image. Doing so prevents the need for common dependencies to be downloaded during application assembly time, speeding up application image builds. It will also simplify the work required by application developers to ensure all of their dependencies are met.
```

Comment:
[Minor suggestion, not required]
It will also simplify..... ==>> It also simplifies.....